up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -41,331 +41,300 @@ service uDiscovery {
-  // 4. code.PERMISSION_DENIED: The caller does not have permission to perform the query
-  rpc LookupUri(v1.UUri) returns (LookupUriResponse) {
-    option (method_id) = 1;
-  }
+  /* 
+  UEs shall call RegisterUE API to add themselves to discovery.
+ */
 
-  // Update a node in the database.
-  // What is returned is the status of the API request as uprotocol.v1.UStatus.
-  // **NOTE:** You MUST have write permission to the node in the database
-  rpc UpdateNode(UpdateNodeRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 2;
-  }
+  /* Example: 
+  RegisterUE(“//ultifi.gm.com/core.example/2.0")
+  Response: Status: code.OK
+ */
 
-  // Query the database to find Node(s). What is passed is the search criterial in
-  // the FindNodeRequest message. the returned FindNodeResponse contains the
-  // resulting query node(s) and their child nodes, as well as the
-  // uprotocol.v1.UStatus for the query. Below are some example queries:
-  // 1. uDomain: `//*.device/`
-  // 2. uDevice: `//device`
-  // 3. uService: `//device.domain/body.access/`
-  // 4. uResource: `//device.domain/body.access//door.door_front_left`
-  // **NOTE:** You MUST have read permission to the node in the database
-  rpc FindNodes(FindNodesRequest) returns (FindNodesResponse) {
-    option (method_id) = 3;
+  rpc RegisterUE(RegisterUERequest) returns ( uprotocol.v1.UStatus){
+    option (method_id) = 1;
   }
 
-  // Query the database to fetch a list of 1 or more properties for a given node.
-  rpc FindNodeProperties(FindNodePropertiesRequest) returns (FindNodePropertiesResponse) {
-    option (method_id) = 4;
-  }
+    /* 
+  UEs shall call UnregisterUE API to remove themselves from discovery.
+ */
 
-  // Remove one or more nodes (and all its children nodes) in the database.
-  // What is returned is the status of the API request as uprotocol.v1.UStatus.
-  // **NOTE:** You MUST have write permission to the deleted all the nodes passed,
-  // all the children nodes, as well as write permission to the parent otherwise
-  // you will get PERMISSION_DENIED and no nodes will be deleted.
-  rpc DeleteNodes(DeleteNodesRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 5;
-  }
+  /* Example: 
+  UnregisterUE(“//ultifi.gm.com/core.example/2.0")
+  Response: Status: code.OK
+ */
 
+ rpc UnregisterUE(UnregisterUERequest) returns ( uprotocol.v1.UStatus){
+  option (method_id) = 1;
+}
 
-  // Add one of more nodes to a parent node. If one of the nodes already exists, this API will return
-  // ALREADY_EXISTS and none of the nodes shall be added to the parent.
-  // **NOTE:** You MUST have write permission to the parent node
-  rpc AddNodes(AddNodesRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 6;
-  }
 
+  /* 
+  UEs shall call SetUETopics API to update discovery with the list of topics they service.
+  This will be used in two scenarios: 
+  1. when a UE is added on a device 
+  2. when UE wants to udpate the topic list
```

Comment:
Perhaps we register and pass the list of all the configured topics in one command.